File: add_classic_course.py
import firebase_admin
from firebase_admin import credentials, firestore
import datetime

# تهيئة Firebase
# ملاحظة: في بيئة Manus، يتم استخدام بيانات الاعتماد الافتراضية أو المسار المحدد
try:
    # محاولة الحصول على بيانات الاعتماد من الملف الموجود في المشروع إذا كان متاحاً
    # أو استخدام الإعدادات الافتراضية للبيئة
    cred = credentials.ApplicationDefault()
    firebase_admin.initialize_app(cred)
except Exception as e:
    print(f"Firebase initialization error: {e}")
    print("Please ensure Firebase is configured correctly in the environment.")

db = firestore.client()

def add_classic_course():
    course_data = {
        "nameAr": "الدورة الكلاسيك",
        "nameEn": "Classic Course",
        "descriptionAr": "دورة شاملة في أساسيات التداول الكلاسيكي، تغطي التحليل الفني وإدارة المخاطر.",
        "descriptionEn": "A comprehensive course in classic trading basics, covering technical analysis and risk management.",
        "type": "free",
        "imageUrl": "/assets/coach_mustafa-cn55cHri.jpg", # استخدام صورة الكوتش مصطفى
        "duration": "4 weeks",
        "sheetUrl": "https://script.google.com/macros/s/AKfycbzU7giZJy_k4nWfvkU1k3qrA8TjRoWFmk23q6dHsbDfZ8WabiBvArtl4tIQAwtvdAPPqQ/exec",
        "telegramUrl": "https://t.me/+CADMeIVdMDQ2Nzg0",
        "whatsappUrl": "",
        "instagramUrl": "",
        "emailUrl": "",
        "features": [
            {"ar": "مجاني بالكامل", "en": "Completely Free"},
            {"ar": "دعم فني مستمر", "en": "Continuous Support"},
            {"ar": "تطبيقات عملية", "en": "Practical Applications"}
        ],
        "enrolledCount": 0,
        "createdAt": datetime.datetime.now(),
        "updatedAt": datetime.datetime.now()
    }

    try:
        # إضافة الكورس إلى مجموعة 'courses'
        doc_ref = db.collection('courses').document('classic_course')
        doc_ref.set(course_data)
        print("Classic Course added successfully!")
    except Exception as e:
        print(f"Error adding course: {e}")

if __name__ == "__main__":
    # هذا السكربت يحتاج لبيئة بايثون مهيأة ببيانات اعتماد Firebase
    # سأقوم بدلاً من ذلك بإنشاء مكون React يقوم بهذه المهمة لمرة واحدة عند فتحه من قبل الأدمن
    print("Script prepared. Will be integrated into Admin Dashboard.")

File: add_classic_course.js
/**
 * سكريبت لإضافة الكورس الكلاسيكي المجاني إلى Firebase
 * يتم تشغيله مرة واحدة لإضافة البيانات
 */

import { initializeApp } from 'firebase/app';
import { getFirestore, collection, doc, setDoc, serverTimestamp } from 'firebase/firestore';
import coachMustafaImg from './src/assets/coach_mustafa.jpg';

// Firebase configuration (استخدم نفس الإعدادات من firebase.js)
const firebaseConfig = {
  // سيتم ملؤها من ملف .env
};

const app = initializeApp(firebaseConfig);
const db = getFirestore(app);

const classicCourse = {
  id: 'classic-trading-course',
  nameAr: 'كورس التداول الكلاسيكي',
  nameEn: 'Classic Trading Course',
  nameFr: 'Cours de Trading Classique',
  descriptionAr: 'كورس شامل للمبتدئين يغطي أساسيات التداول والتحليل الفني والأساسي. تعلم من الكوتش مصطفى كيف تبدأ رحلتك في عالم التداول بثقة واحترافية.',
  descriptionEn: 'Comprehensive beginner course covering trading basics, technical and fundamental analysis. Learn from Coach Mustafa how to start your trading journey with confidence and professionalism.',
  descriptionFr: 'Cours complet pour débutants couvrant les bases du trading, l\'analyse technique et fondamentale. Apprenez du Coach Mustafa comment commencer votre parcours de trading avec confiance et professionnalisme.',
  type: 'free',
  imageUrl: coachMustafaImg,
  duration: '8 أسابيع / 8 Weeks / 8 Semaines',
  price: 0,
  enrolledCount: 0,
  features: {
    ar: [
      'محتوى شامل للمبتدئين',
      'دروس فيديو عالية الجودة',
      'تمارين عملية وتطبيقات',
      'شهادة إتمام مجانية',
      'دعم فني مستمر',
      'وصول مدى الحياة'
    ],
    en: [
      'Comprehensive beginner content',
      'High-quality video lessons',
      'Practical exercises and applications',
      'Free completion certificate',
      'Continuous technical support',
      'Lifetime access'
    ],
    fr: [
      'Contenu complet pour débutants',
      'Leçons vidéo de haute qualité',
      'Exercices pratiques et applications',
      'Certificat de fin gratuit',
      'Support technique continu',
      'Accès à vie'
    ]
  },
  curriculum: {
    ar: [
      'مقدمة في عالم التداول',
      'أساسيات التحليل الفني',
      'التحليل الأساسي',
      'إدارة المخاطر',
      'استراتيجيات التداول',
      'علم النفس التجاري',
      'منصات التداول',
      'التطبيق العملي'
    ],
    en: [
      'Introduction to Trading',
      'Technical Analysis Basics',
      'Fundamental Analysis',
      'Risk Management',
      'Trading Strategies',
      'Trading Psychology',
      'Trading Platforms',
      'Practical Application'
    ],
    fr: [
      'Introduction au Trading',
      'Bases de l\'Analyse Technique',
      'Analyse Fondamentale',
      'Gestion des Risques',
      'Stratégies de Trading',
      'Psychologie du Trading',
      'Plateformes de Trading',
      'Application Pratique'
    ]
  },
  instructor: {
    name: 'Coach Mustafa',
    nameAr: 'الكوتش مصطفى',
    nameFr: 'Coach Mustafa',
    bio: {
      ar: 'خبير تداول معتمد مع أكثر من 10 سنوات من الخبرة في الأسواق المالية',
      en: 'Certified trading expert with over 10 years of experience in financial markets',
      fr: 'Expert en trading certifié avec plus de 10 ans d\'expérience sur les marchés financiers'
    },
    image: coachMustafaImg
  },
  createdAt: serverTimestamp(),
  updatedAt: serverTimestamp(),
  status: 'active',
  isClassic: true
};

async function addClassicCourse() {
  try {
    const courseRef = doc(db, 'courses', 'classic-trading-course');
    await setDoc(courseRef, classicCourse);
    console.log('✅ تم إضافة الكورس الكلاسيكي بنجاح!');
  } catch (error) {
    console.error('❌ خطأ في إضافة الكورس:', error);
  }
}

// تشغيل السكريبت
addClassicCourse();
